tests/test_basic_auth.py

```diff
@@ -0,0 +1,143 @@
+import os
+import socket
+from urllib import parse as url_parse
+
+import requests
+import pytest
+import requests_mock
+
+from domino import Domino, bearer_auth
+
+
+def domino_is_reachable(url=os.getenv('DOMINO_HOST_KEY_NAME'), port='443'):
+    """
+    Confirm that a deployment is acessible for tests that require it.
+    """
+    if url is None:
+        return False
+
+    try:
+        fqdn = url_parse.urlsplit(url).netloc
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.connect((fqdn, int(port)))
+        is_reachable = True
+    except socket.gaierror:
+        print(f"{fqdn}:{port} is not reachable")
+        is_reachable = False
+    finally:
+        s.close()
+
+    return is_reachable
+
+
+@pytest.fixture
+def mock_domino_version_response():
+    """
+    Simulate a valid repsonse to an authenticated /version GET request.
+    """
+    version_info = {
+        "buildId": "12345",
+        "buildUrl": "https://some.server.com/domino/12345",
+        "commitId": "123deadbeef456",
+        "commitUrl": "https://some.repo.com/domino/commit/123deadbeef456",
+        "timestamp": "2021-06-14T16:50:02Z",
+        "version": "9.9.9"
+    }
+
+    with requests_mock.Mocker() as mock_endpoint:
+        mock_endpoint.get("http://mock.domino.com/version", json=version_info, status_code=200)
+        yield
+
+
+@pytest.fixture
+def dummy_token_file(tmpdir):
+    """
+    Simulate a token file.
+    """
+    token_file = tmpdir.join("dummy_token_file.txt")
+    token_file.write('top_secret_auth_token')
+    return token_file
+
+
+@pytest.fixture
+def real_token_file(tmpdir):
+    """
+    Derive a real authentication token (assumes that certain packages are available).
```

Comment:
Does it make sense to bring these packages in? Trying to understand long term strategy here.

Reply:
It's an internal package that's a) still under development, and b) probably won't be made available for public consumption anyway. The thinking is that I wanted these tests to be available for internal use, but not cause outright failures (i.e., false negatives) in case all of the test requirements aren't available if by an external contributor.

Here's an example of what the results look like without sample skipped and xfail'ed tests.
```
tests/test_basic_auth.py::test_object_creation_with_api_key PASSED
tests/test_basic_auth.py::test_object_creation_with_token_file PASSED
tests/test_basic_auth.py::test_auth_against_real_deployment_with_api_key SKIPPED (No API key in environment)
tests/test_basic_auth.py::test_auth_against_real_deployment_with_token_file XFAIL (Could not derive valid token for token file)
```